x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/TimeSeriesSortedSourceOperatorFactory.java

```diff
@@ -161,10 +158,9 @@ public Page getOutput() {
                 timestampIntervalBuilder = blockFactory.newLongVectorBuilder(Math.min(remainingDocs, maxPageSize));
                 tsids = tsOrdBuilder.build();
                 tsOrdBuilder = blockFactory.newIntVectorBuilder(Math.min(remainingDocs, maxPageSize));
-
                 page = new Page(
                     currentPagePos,
-                    new DocVector(shard.asVector(), leaf, docs, singleSegmentNonDecreasing).asBlock(),
+                    new DocVector(shard.asVector(), leaf, docs, leaf.isConstant()).asBlock(),
```

Comment:
If we load documents from a single segment, then we can set `singleSegmentNonDecreasing` to true.